src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -703,6 +727,38 @@ private boolean showDeleteConfirmationDialog(int numberOfEntries) {
         }
     }
 
+    /**
+     * Initializes a pop-up dialog box to confirm whether the user wants to delete attached files from selected entry
+     * Keep track of user preference:
+     * if the user prefers always delete attached files, delete the files without displaying the dialog box
+     *
+     * @param linkedFileList A non-duplicated list of LinkedFile to be deleted
+     * @return true if user confirm to delete attached files
+     */
+    private boolean showLinkedFileDeleteConfirmationDialog(List<LinkedFile> linkedFileList) {
+        if (preferencesService.getFilePreferences().alwaysDeleteLinkedFile()) {
+            return true;
+        } else {
+            String title = Localization.lang("Delete attached file(s)");
+            String message = linkedFileList.stream()
+                                           .map(linkedFile -> linkedFile.findIn(bibDatabaseContext, preferencesService.getFilePreferences())
+                                                                        .map(Path::getFileName)
+                                                                        .map(Path::toString)
+                                                                        .orElse(""))
+                                           .collect(Collectors.joining("\n", Localization.lang("Following attached file(s) will be deleted") + "\n\n", ""));
+            String okButton = Localization.lang("Delete attached files");
+            String cancelButton = Localization.lang("Keep attached files");
+
+            return dialogService.showConfirmationDialogWithOptOutAndWait(
+                    title,
+                    message,
+                    okButton,
+                    cancelButton,
+                    Localization.lang("Always delete attached file(s)"),
+                    optOut -> preferencesService.getFilePreferences().setAlwaysDeleteLinkedFile(optOut));
+        }
```

Comment:
Replace `} else {` by `}` and move that thing one level uip

Reason: Fail fast, keep branch length short.